### PR TITLE
Choose between npm and yarn when needed

### LINKED ISSUE #14: NPM and Yarn Slapfight
It looks like this buildpack is hardcoded to use NPM instead of Yarn. We ran into a problem where Yarn was installing and respecting it's lockfile, but then NPM was coming through and installing without a lockfile, thereby overwriting some locked versions.

For the moment we just forked and removed the NPM install line. What would be the best course of action for detecting whether to use Yarn or NPM?

### PR DESCRIPTION
**Pull request by brunoarueira**
Fixes #14

### CODE REVIEW

**Comment by febeling** [comment]
Thanks for the PR! Sorry for the late response, I got distracted. 

Would you add a section to the README about Usage for the `YARN` environment variable?

File: bin/compile [hunk 1]
```diff
@@ -14,6 +14,15 @@ env_dir=$3
 
 cd $build_dir
 
+[ -f "$build_dir/yarn.lock" ] && YARN=true || YARN=false
```
Comment (febeling) [inline bin/compile]: @brunoarueira could you unpack this line for me? I can't decide if it's supposed to be an assignment or a condition test...

**Review by febeling — COMMENTED** [review]
Looks great overall. I have this one question, see below.

**Comment by brunoarueira** [comment]
@febeling done 👍

**Comment by febeling** [comment]
@brunoarueira so awesome! Thanks for the PR 👏

**Pull request merged**